analyzers/tests/SonarAnalyzer.UnitTest/TestFramework/OldVerifier.cs

```diff
@@ -301,68 +302,104 @@ public static void VerifyNoExceptionThrown(string path, DiagnosticAnalyzer[] dia
                 .AddPaths(RemoveTestCasesPrefix(paths))
                 .Verify();
 
+        [Obsolete("Use VerifierBuilder instead.")]
         public static void VerifyAnalyzer(string path,
                                           DiagnosticAnalyzer[] diagnosticAnalyzers,
-                                          IEnumerable<ParseOptions> options = null,
+                                          ImmutableArray<ParseOptions> options = default,
                                           CompilationErrorBehavior checkMode = CompilationErrorBehavior.Default,
                                           OutputKind outputKind = OutputKind.DynamicallyLinkedLibrary,
                                           IEnumerable<MetadataReference> additionalReferences = null,
                                           DiagnosticDescriptor[] onlyDiagnostics = null) =>
-            VerifyAnalyzer(new[] { path }, diagnosticAnalyzers, options, checkMode, outputKind, additionalReferences, null, onlyDiagnostics);
+            diagnosticAnalyzers.Aggregate(new VerifierBuilder(), (builder, analyzer) => builder.AddAnalyzer(() => analyzer))
+                .AddReferences(additionalReferences)
+                .AddPaths(RemoveTestCasesPrefix(path))
+                .WithOptions(options.IsDefault ? ImmutableArray<ParseOptions>.Empty : options)
+                .WithErrorBehavior(checkMode)
+                .WithOutputKind(outputKind)
+                .WithOnlyDiagnostics(onlyDiagnostics ?? Array.Empty<DiagnosticDescriptor>())
+                .Verify();
 
+        [Obsolete("Use VerifierBuilder instead.")]
         public static void VerifyAnalyzer(string path,
                                           DiagnosticAnalyzer diagnosticAnalyzer,
                                           IEnumerable<MetadataReference> additionalReferences,
                                           string sonarProjectConfigPath) =>
-            VerifyAnalyzer(new[] { path },
-                           new[] { diagnosticAnalyzer },
-                           null, CompilationErrorBehavior.Default,
-                           OutputKind.DynamicallyLinkedLibrary,
-                           additionalReferences,
-                           sonarProjectConfigPath);
+            new VerifierBuilder()
+                .AddAnalyzer(() => diagnosticAnalyzer)
+                .AddReferences(additionalReferences)
+                .AddPaths(RemoveTestCasesPrefix(path))
+                .WithSonarProjectConfigPath(sonarProjectConfigPath)
+                .Verify();
```

Comment:
pedantic: We usually make the first method invocation on the same line and then align the rest of the calls with the first one.

Reply:
While I usually prefer less lines, I find this alignment easier to understand in this case. All modifications start on the same indent here.

Reply:
I now better see what you meant based on this comment:
https://github.com/SonarSource/sonar-dotnet/pull/5213#discussion_r774479132